src/sharing/ShareWrapper.js

```diff
@@ -170,6 +172,36 @@ class ShareWrapper extends React.Component<Props, State> {
     }
   };
 
+  deleteItem = toDelete => {
+    const files = [...this.state.files];
+    const filteredItems = files.filter(item => item.url !== toDelete.url);
+    this.setState({ files: filteredItems });
+    if (filteredItems.length === 0) {
+      showToast('Cancelled Share');
+      this.onShareCancelled();
+    }
```

Comment:
And then I guess this needs to go inside that `setState` callback too.

The length check should then also check that the old length was nonzero -- so if these get repeated for some reason, we only do the cancel once.

And the `showToast` and `onShareCancelled` should probably get put inside a `setTimeout` -- so they don't happen synchronously inside the state update. Maybe that'd be fine, but it feels like the sort of thing that might be glitchy.